freeipa/src/main/java/com/sequenceiq/freeipa/controller/RecipeV1Controller.java

```diff
@@ -0,0 +1,50 @@
+package com.sequenceiq.freeipa.controller;
+
+import static com.sequenceiq.authorization.resource.AuthorizationResourceAction.DESCRIBE_RECIPE;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceAction.EDIT_ENVIRONMENT;
+import static com.sequenceiq.authorization.resource.AuthorizationVariableType.CRN;
+import static com.sequenceiq.authorization.resource.AuthorizationVariableType.NAME_LIST;
+
+import javax.inject.Inject;
+import javax.validation.Valid;
+
+import org.springframework.stereotype.Controller;
+
+import com.sequenceiq.authorization.annotation.CheckPermissionByRequestProperty;
+import com.sequenceiq.authorization.annotation.RequestObject;
+import com.sequenceiq.freeipa.api.v1.recipe.RecipeV1Endpoint;
+import com.sequenceiq.freeipa.api.v1.recipe.model.RecipeAttachDetach;
+import com.sequenceiq.freeipa.service.recipe.FreeIpaRecipeService;
+import com.sequenceiq.freeipa.service.stack.StackService;
+import com.sequenceiq.freeipa.util.CrnService;
+
+@Controller
+public class RecipeV1Controller implements RecipeV1Endpoint {
+
+    @Inject
+    private FreeIpaRecipeService freeIpaRecipeService;
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private CrnService crnService;
+
+    @Override
+    @CheckPermissionByRequestProperty(path = "environmentCrn", type = CRN, action = EDIT_ENVIRONMENT)
+    @CheckPermissionByRequestProperty(path = "recipes", type = NAME_LIST, action = DESCRIBE_RECIPE)
+    public void attachRecipes(@Valid @RequestObject RecipeAttachDetach recipeAttach) {
+        String accountId = crnService.getCurrentAccountId();
+        Long stackId = stackService.getIdByEnvironmentCrnAndAccountId(recipeAttach.getEnvironmentCrn(), accountId);
```

Comment:
Is there any reason why this logic is not in the FreeIpaRecipeService? I like to keep Controllers as simple as possible.